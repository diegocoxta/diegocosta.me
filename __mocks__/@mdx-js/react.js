module.exports = {
  MDXProvider: jest.fn(({ children }) => children),
};
