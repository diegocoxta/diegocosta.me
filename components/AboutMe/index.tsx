import Container from '~/components/Container';

import styles from './styles.module.css';

interface AboutMeProps {
  bio: string;
  links?: Array<{
    url: string;
    label: string;
  }>;
}

export default function AboutMe(props: AboutMeProps) {
  return (
    <Container>
      {props.bio.split('\n').map((p: string) => (
        <p className={styles.paragraph} data-testid="about-me-bio" key={p} dangerouslySetInnerHTML={{ __html: p }} />
      ))}
      {props.links && (
        <ul className={styles.links} data-testid="about-me-links">
          {props.links.map((nav, index) => (
            <li className={styles.linksItem} key={`nav-${index}`} data-testid="about-me-links-item">
              <a className={styles.linksLink} href={nav.url} rel="me" target="_blank">
                {nav.label}
              </a>
            </li>
          ))}
        </ul>
      )}
    </Container>
  );
}
