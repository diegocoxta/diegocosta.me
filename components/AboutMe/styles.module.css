.paragraph {
  font-size: 20px;
  line-height: 1.4;
  color: var(--main-text-color);

  @media (width >= 760px) {
    font-size: 24px;
  }
}

.links {
  margin: 0;
  padding: 0;

  @media (width >= 760px) {
    display: flex;
  }
}

.linksItem {
  list-style: none;
  font-size: 22px;
  font-weight: 700;
  margin: 0 24px 5px 0;

  @media (width >= 760px) {
    margin: 0 24px 0 0;
  }
}

.linksLink {
  text-decoration: none;
  color: var(--main-text-color);
  display: flex;
  padding: 0;
  text-transform: uppercase;

  &::after {
    content: '.';
    display: block;
    color: var(--main-background-color);
    font-size: 38px;
    line-height: 0.5;

    @media (width >= 760px) {
      line-height: 0.4;
    }
  }

  &:hover::after {
    color: var(--main-text-color);
  }
}
