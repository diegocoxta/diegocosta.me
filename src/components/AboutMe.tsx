import React from 'react';
import styled from 'styled-components';

const Container = styled.div``;

const Paragraph = styled.p`
  font-size: 24px;
  line-height: 1.4;
  color: ${({ theme }) => theme.textColor};
`;

const Links = styled.ul`
  margin: 0;
  padding: 0;

  @media (min-width: 760px) {
    display: flex;
  }
`;

const LinksItem = styled.li`
  list-style: none;
  font-size: 24px;
  font-weight: 700;
  margin: 0 24px 5px 0;

  @media (min-width: 760px) {
    font-size: 22px;
    margin: 0 24px 0 0;
  }
`;

const LinksLink = styled.a`
  text-decoration: none;
  color: ${({ theme }) => theme.textColor};
  display: flex;
  padding: 0;
  text-transform: uppercase;

  &:after {
    content: '.';
    display: block;
    color: ${({ theme }) => theme.backgroundColor};
    font-size: 38px;
    line-height: 0.5;

    @media (min-width: 760px) {
      line-height: 0.4;
    }
  }

  &:hover:after {
    color: ${({ theme }) => theme.textColor};
  }
`;

export interface AboutMeProps {
  bio?: string;
  socialLinks?: [
    {
      label: string;
      url: string;
      rel?: string;
    },
  ];
}

export default function AboutMe(props: AboutMeProps) {
  return (
    <Container>
      {props.bio
        ?.split('\n')
        .map((p: string) => <Paragraph data-testid="about-me-bio" key={p} dangerouslySetInnerHTML={{ __html: p }} />)}
      {props.socialLinks && (
        <Links data-testid="about-me-sociallinks">
          {props.socialLinks.map((nav, index) => (
            <LinksItem key={`nav-${index}`} data-testid="about-me-sociallinks-item">
              <LinksLink href={nav.url} rel={nav.rel} target="_blank">
                {nav.label}
              </LinksLink>
            </LinksItem>
          ))}
        </Links>
      )}
    </Container>
  );
}
