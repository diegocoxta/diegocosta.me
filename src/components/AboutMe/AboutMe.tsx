import React from 'react';
import styled from 'styled-components';
import { MDXRenderer } from 'gatsby-plugin-mdx';

import Container from '~/components/Container';

const Content = styled.div`
  @media (min-width: 760px) {
    padding-bottom: 40px;
  }

  p {
    font-size: 18px;
    line-height: 1.4;
    color: ${({ theme }) => theme.textColor};

    @media (min-width: 760px) {
      font-size: 24px;
    }
  }

  ul {
    margin: 0;
    padding: 0;

    @media (min-width: 760px) {
      display: flex;
    }
  }

  li {
    list-style: none;
    font-size: 18px;
    font-weight: 900;
    margin: 0 24px 5px 0;

    @media (min-width: 760px) {
      font-size: 20px;
      margin: 0 24px 0 0;
    }

    a {
      text-decoration: none;
      color: ${({ theme }) => theme.textColor};
      display: flex;
      padding: 0;
      text-transform: uppercase;

      :after {
        content: '.';
        display: block;
        color: ${({ theme }) => theme.backgroundColor};
        font-size: 38px;
        line-height: 0.2;

        @media (min-width: 760px) {
          line-height: 0.3;
        }
      }

      :hover:after {
        color: ${({ theme }) => theme.textColor};
      }
    }
  }
`;

interface AboutMeProps {
  mdxContent: string;
}

export default function AboutMe(props: AboutMeProps): JSX.Element {
  return (
    <Container>
      <Content>
        <MDXRenderer>{props.mdxContent}</MDXRenderer>
      </Content>
    </Container>
  );
}
