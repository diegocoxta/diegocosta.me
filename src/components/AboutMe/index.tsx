import React from 'react';
import { useStaticQuery, graphql } from 'gatsby';

import AboutMe from './AboutMe';

export default (): React.ReactElement => {
  const {
    site: {
      siteMetadata: { aboutMe },
    },
  } = useStaticQuery(
    graphql`
      query {
        site {
          siteMetadata {
            aboutMe
          }
        }
      }
    `
  );

  return <AboutMe paragraphs={aboutMe} />;
};
