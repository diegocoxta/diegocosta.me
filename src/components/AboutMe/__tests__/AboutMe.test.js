import React from 'react';
import { render } from '@testing-library/react';

import AboutMe from '../AboutMe';

describe('<AboutMe />', () => {
  it('should render properly', () => {
    const { baseElement } = render(
      <AboutMe>
        <p>You can hear more about here:</p>
      </AboutMe>
    );

    expect(baseElement).toMatchSnapshot();
  });
});
