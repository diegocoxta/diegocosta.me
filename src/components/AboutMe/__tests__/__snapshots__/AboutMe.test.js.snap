// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutMe /> should render properly 1`] = `
.c0 {
  max-width: 800px;
  margin: 0 auto;
  padding: 0 20px;
}

.c1 p {
  font-size: 18px;
  line-height: 1.4;
}

.c1 ul {
  margin: 0;
  padding: 0;
}

.c1 li {
  list-style: none;
  font-size: 18px;
  font-weight: 900;
  margin: 0 24px 5px 0;
}

.c1 li a {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  text-transform: uppercase;
}

.c1 li a:after {
  content: '.';
  display: block;
  font-size: 38px;
  line-height: 0.2;
}

@media (min-width:760px) {
  .c1 {
    padding-bottom: 40px;
  }
}

@media (min-width:760px) {
  .c1 p {
    font-size: 24px;
  }
}

@media (min-width:760px) {
  .c1 ul {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:760px) {
  .c1 li {
    font-size: 20px;
    margin: 0 24px 0 0;
  }
}

@media (min-width:760px) {
  .c1 li a:after {
    line-height: 0.3;
  }
}

<body>
  <div>
    <section
      class="c0"
    >
      <main>
        <div
          class="c1"
        >
          <p>
            You can hear more about here:
          </p>
        </div>
      </main>
    </section>
  </div>
</body>
`;
